omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,217 @@
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isAutomaticGenerationEnabled());
+		assertFalse(identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isManualEntryEnabled());
+	}
+	
+	@Test(expected = org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException.class)
+	public void shouldNotEditAnAutoGenerationOptionProvidedWithAllNullParams() throws Exception {
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("location", null);
+		autoGenerationOption.add("source", null);
+		autoGenerationOption.add("manualEntryEnabled", null);
+		autoGenerationOption.add("automaticGenerationEnabled", null);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		assertEquals(LOCATION,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getLocation().getUuid());
+		assertEquals(SOURCE,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getSource().getUuid());
+		assertTrue(
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isAutomaticGenerationEnabled());
+		assertTrue(identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isManualEntryEnabled());
+	}
+	
+	@Test
+	public void shouldEditAnAutoGenerationOptionByAnyNumberOFArguments() throws Exception {
+		boolean automaticGenerationEnabled = false;
+		boolean manualEntryEnabled = false;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		assertFalse(
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isAutomaticGenerationEnabled());
+		assertFalse(identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isManualEntryEnabled());
+	}
+	
+	@Test
+	public void shouldPurgeAnAutoGenerationOption() throws Exception {
+		boolean automaticGenerationEnabled = true;
+		boolean manualEntryEnabled = true;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req1 = request(RequestMethod.POST, getURI());
```

Comment:
Why do you first add a new autogeneration option?
I thought this test was only about purging instead of adding a new one.